netket/operator/_sumoperators.py

```diff
@@ -40,6 +40,21 @@ def __init__(
             coefficients: A coefficient for each ContinuousOperator object
             dtype: Data type of the matrix elements. Defaults to `np.float64`
         """
+        if len(operators) != len(coefficients):
+            raise AssertionError("Each operator needs a coefficient")
```

Comment:
I think that for ease of use it would be useful to accept a scalar value for `coefficients` (as documented by the signature) and upscale it to the right length

```suggestion
        if is_scalar(coefficients):
            coefficients = [coefficients for _ in operators]

        if len(operators) != len(coefficients):
            raise AssertionError("Each operator needs a coefficient")
```

you will also need to import
```
from netket.utils import is_scalar
```